SwiftWisdom/SwiftWisdom/Core/UIKit/UIView+GetConstraints.swift

```diff
@@ -59,27 +59,32 @@ public extension UIView {
     }
     
     /**
-    More precise way of getting constraints for a view. The constraint does not have to belong to the reciever -- this method searches the superview for the matching constraint as well.
-    
-    :param: attribute     Attribute to look for belonging to the receiver
-    :param: toItem        Probably a UIView, that the receiver is related to.
-    :param: itemAttribute Attribute that should be used with the toItem
-    
-    :returns: The first constraint that matches. May return unexpected constraint if receiver contains multiple constraints with this item and itemAttribute.
-    */
-    public func ip_constraintForAttribute(attribute: NSLayoutAttribute, onView view1: UIView? = self, toView view2: UIView? = nil, viewAttribute: NSLayoutAttribute? = nil) -> NSLayoutConstraint? {
+     More precise way of getting constraints for a view. The constraint does not have to belong to the reciever -- this method searches the superview for the matching constraint as well.
+     
+     :param: attribute     Attribute to look for belonging to the receiver
+     :param: view1        Defaults to the receiver, but could be any subview as well.
+     :param: view2        Probably a UIView, that view1 is related to.
+     :param: viewAttribute Attribute that should be used with the toItem
+     
+     :returns: The first constraint that matches. May return unexpected constraint if receiver contains multiple constraints with this item and itemAttribute.
+     */
+    public func ip_constraintForAttribute(attribute: NSLayoutAttribute, var onView view1: UIView? = nil, toView view2: UIView? = nil, viewAttribute: NSLayoutAttribute? = nil) -> NSLayoutConstraint? {
+        
+        if view1 == nil {
+            view1 = self
+        }
         
-        if let toAttribute = itemAttribute, let toItem = view2 {
+        if let toAttribute = viewAttribute, let toItem = view2 {
             
             return constraints.filter { constraint in
-                return constraint.ip_relatesView(view: view1, viaAttribute: attribute, toView: toItem, andItsAttribute: toAttribute)
+                return constraint.ip_relatesView(view: view1!, viaAttribute: attribute, toView: toItem, andItsAttribute: toAttribute)
```

Comment:
Avoid unwrapping in this way in general, save `!` unwrapping only for a last resort.  Even though the current implementation, we can say that it will definitely exist, we shouldn't rely on it by human code comprehension.  We should let Swift handle it for us, so it can't reach this line of code unless `onView` (`view1`) has value.  This allows us to rely on its implementation here regardless of what happens above in future implementations since it won't build unless that condition is satisfied.